### ISSUE TITLE
False positive error with variance validation in protocol

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6883]
The following code should not generate a type error, but it currently does.

```python
from typing import Protocol, TypeVar

T1_contra = TypeVar("T1_contra", contravariant=True)

class Protocol13(Protocol[T1_contra]):  # OK
    # "self" and "cls" parameters are exempt from variance calculations.
    def m1(self: "Protocol13[T1_contra]", x: T1_contra) -> None:
        ...

    @classmethod
    def m2(cls: "type[Protocol13[T1_contra]]") -> None:
        ...
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.346, which I just published. It will also be included in a future release of pylance.
